library/src/hcc_detail/hipsolver.cpp

```diff
@@ -173,6 +173,227 @@ catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverCungqr_bufferSize(hipsolverHandle_t handle,
+                                             int               m,
+                                             int               n,
+                                             int               k,
+                                             hipsolverComplex* A,
+                                             int               lda,
+                                             hipsolverComplex* tau,
+                                             int*              lwork)
+try
+{
+    size_t sz;
+
+    rocblas_start_device_memory_size_query((rocblas_handle)handle);
+    rocblas_status status
+        = rocsolver_cungqr((rocblas_handle)handle, m, n, k, nullptr, lda, nullptr);
+    rocblas_stop_device_memory_size_query((rocblas_handle)handle, &sz);
+
+    if(sz > INT_MAX)
+        return HIPSOLVER_STATUS_INTERNAL_ERROR;
+
+    *lwork = (int)sz;
+    return rocblas2hip_status(status);
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverZungqr_bufferSize(hipsolverHandle_t       handle,
+                                             int                     m,
+                                             int                     n,
+                                             int                     k,
+                                             hipsolverDoubleComplex* A,
+                                             int                     lda,
+                                             hipsolverDoubleComplex* tau,
+                                             int*                    lwork)
+try
+{
+    size_t sz;
+
+    rocblas_start_device_memory_size_query((rocblas_handle)handle);
+    rocblas_status status
+        = rocsolver_zungqr((rocblas_handle)handle, m, n, k, nullptr, lda, nullptr);
+    rocblas_stop_device_memory_size_query((rocblas_handle)handle, &sz);
+
+    if(sz > INT_MAX)
+        return HIPSOLVER_STATUS_INTERNAL_ERROR;
+
+    *lwork = (int)sz;
+    return rocblas2hip_status(status);
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverSorgqr(hipsolverHandle_t handle,
+                                  int               m,
+                                  int               n,
+                                  int               k,
+                                  float*            A,
+                                  int               lda,
+                                  float*            tau,
+                                  float*            work,
+                                  int               lwork,
+                                  int*              devInfo)
+try
+{
+    if(work != nullptr)
+        rocblas_set_workspace((rocblas_handle)handle, work, lwork);
+    else
+    {
+        if(!rocblas_is_managing_device_memory((rocblas_handle)handle))
+            rocblas_set_workspace((rocblas_handle)handle, nullptr, 0);
+    }
```

Comment:
It's rather trivial, but would `else if` make sense here?

Reply:
That would look nicer, I think. It'll affect all the other functions too, so I'll probably wait until they're all in and open a new PR. Should be able to do a search-and-replace.